pkg/apis/seedmanagement/types_managedseed.go

```diff
@@ -0,0 +1,161 @@
+
+// Gardenlet specifies gardenlet deployment parameters and the GardenletConfiguration used to configure gardenlet.
+type Gardenlet struct {
+	// Deployment specifies certain gardenlet deployment parameters, such as the number of replicas,
+	// the image, etc.
+	Deployment *GardenletDeployment
+	// Config is the GardenletConfiguration used to configure gardenlet.
+	Config *runtime.RawExtension
+	// Bootstrap is the mechanism that should be used for bootstrapping gardenlet connection to the Garden cluster. One of ServiceAccount, BootstrapToken, None.
+	// If set to ServiceAccount or BootstrapToken, a service account or a bootstrap token will be created in the garden cluster and used to compute the bootstrap kubeconfig.
+	// If set to None, the gardenClientConnection.kubeconfig field will be used to connect to the Garden cluster. Defaults to BootstrapToken.
+	Bootstrap *Bootstrap
+	// MergeWithParent specifies whether the deployment parameters and GardenletConfiguration of the parent gardenlet
+	// should be merged with the specified deployment parameters and GardenletConfiguration. Defaults to true.
+	MergeWithParent *bool
+}
+
+// GardenletDeployment specifies certain gardenlet deployment parameters, such as the number of replicas,
+// the image, etc.
+type GardenletDeployment struct {
+	// ReplicaCount is the number of gardenlet replicas. Defaults to 1.
+	ReplicaCount *int32
+	// RevisionHistoryLimit is the number of old gardenlet ReplicaSets to retain to allow rollback. Defaults to 1.
+	RevisionHistoryLimit *int32
+	// ServiceAccountName is the name of the ServiceAccount to use to run gardenlet pods.
+	ServiceAccountName *string
+	// Image is the gardenlet container image.
+	Image *Image
+	// Resources are the compute resources required by the gardenlet container.
+	Resources *corev1.ResourceRequirements
+	// PodLabels are the labels on gardenlet pods.
+	PodLabels map[string]string
+	// PodAnnotations are the annotations on gardenlet pods.
+	PodAnnotations map[string]string
+	// AdditionalVolumes is the list of additional volumes that should be mounted by gardenlet containers.
+	AdditionalVolumes []corev1.Volume
+	// AdditionalVolumeMounts is the list of additional pod volumes to mount into the gardenlet container's filesystem.
+	AdditionalVolumeMounts []corev1.VolumeMount
+	// Env is the list of environment variables to set in the gardenlet container.
+	Env []corev1.EnvVar
+	// VPA specifies whether to enable VPA for gardenlet. Defaults to true.
+	VPA *bool
+}
+
+// Image specifies container image parameters.
+type Image struct {
+	// Repository is the image repository.
+	Repository *string
+	// Tag is the image tag.
+	Tag *string
+	// PullPolicy is the image pull policy. One of Always, Never, IfNotPresent.
+	// Defaults to Always if latest tag is specified, or IfNotPresent otherwise.
+	PullPolicy *corev1.PullPolicy
+}
+
+// Bootstrap describes a mechanism for bootstrapping gardenlet connection to the Garden cluster.
+type Bootstrap string
```

Comment:
```suggestion
type BootstrapMode string
```

Reply:
Is this really necessary? If I change the type name I should probably change the field name as well, and then adapt to this change in 5 other downstream PRs. The difference is very small and I personally prefer `Bootstrap` as it's shorter .

Reply:
I don't insist on it, but I prefer to have the "Mode" suffix because it's simply more clear what this is about. Is it really that much effort to change the other PRs accordingly? You have to rebase them anyways once this PR was merged.

Reply:
From my perspective `BootstrapMode` is neither more nor less clear than `Bootstrap`, they are the same. There is a comment explaining the meaning and the usage, without it both would not be clear. It would take some effort to properly rename it everywhere and I question whether this is really needed. If you don't insist, could we keep it as it is?